src/Microsoft.AspNetCore.Sockets.Client/LongPollingTransport.cs

```diff
@@ -143,10 +146,13 @@ public Task StartAsync(Uri url, IChannelConnection<SendMessage, Message> applica
             _logger.LogInformation("Receive loop stopped");
         }
 
-        private IList<Message> ParsePayload(byte[] payload)
+        private IList<Message> ParsePayload(byte[] payload, MessageFormat messageFormat)
         {
             var reader = new BytesReader(payload);
-            var messageFormat = MessageParser.GetFormat(reader.Unread[0]);
+            if (messageFormat != MessageParser.GetFormatFromIndicator(reader.Unread[0]))
+            {
+                throw new FormatException($"Format indicator '{(char)reader.Unread[0]}' does not match format determined by Content-Type '{MessageFormatter.GetContentType(messageFormat)}'");
```

Comment:
Sure, it shouldn't be too hard. It strikes me that there aren't currently any tests here that deal with actually parsing the poll result.